final-project/application.py

```diff
@@ -41,10 +41,43 @@ def index():
     """Homepage"""
 
     # Keep the user logged in
-    session.get("user_id")
+    user_id = session.get("user_id")
 
-    # TODO
-    return apology("TODO")
+    # TODO: Get the username
+    user_info = db.execute(
+            "SELECT username FROM users WHERE id = :user_id"
+            + " LIMIT 1",
+            user_id=user_id)
+
+    # Ensure user was found
+    if user_info == None:
+        return apology("You were not found in our database. Please login.")
+
+    # Get username of the current user
+    username = user_info[0]['username']
+
+    # Get the most recently added products. Limit to 2.
+    recent_products = db.execute(
+            "SELECT product_name, image"
+            + " FROM products"
+            + " ORDER BY id DESC"
+            + " LIMIT 2")
```

Comment:
Much better! There are also these ways of making multi-line string: https://stackoverflow.com/a/10660443/2204868

Reply:
thanks, I read through the whole python style guide and didn't come across this. I am glad you could find it in a couple seconds haha.

Those multi-line strings took way too long to make in my version. I'll change it now.